src/compose/service.coffee

```diff
@@ -537,27 +534,29 @@ module.exports = class Service
 
 	# TODO: map ports for any of the possible formats "container:host/protocol", port ranges, etc.
 	getPortsAndPortBindings: =>
-		exposedPorts = {}
-		portBindings = {}
+		portMaps = []
 		if @ports?
 			for port in @ports
-				m = port.match(PORTS_REGEX)
-				if m? # Ignore invalid port mappings
-					[ _unused, host = '', external, externalEnd, internal, internalEnd, protocol = 'tcp' ] = m
-					external ?= internal
-					internalEnd ?= internal
-					externalEnd ?= external
-					externalRange = _.map([external..externalEnd], String)
-					internalRange = _.map([internal..internalEnd], String)
-					if externalRange.length == internalRange.length # Ignore invalid port mappings
-						for hostPort, ind in externalRange
-							containerPort = internalRange[ind]
-							exposedPorts["#{containerPort}/#{protocol}"] = {}
-							portBindings["#{containerPort}/#{protocol}"] = [ { HostIp: host, HostPort: hostPort } ]
+				portMaps.push(new PortMap(port))
```

Comment:
This would be better as
```coffee
portMaps =
	for port in @ports
		new PortMap(port)
````
or
```coffee
portMaps = _.map @ports, (port) ->
	new PortMap(port)
```
(with the second you can also drop the `if @ports?` as lodash will handle it)